tests/test_user.py

```diff
@@ -0,0 +1,347 @@
+import pytest
+
+from auth.config import Settings
+from fastapi.testclient import TestClient
+from main import app
+from tests.datagen import AccessTokenPayloadFactory
+
+client = TestClient(app)
+
+
+@pytest.fixture(autouse=True)
+def mock_auth_settings(mocker):
+    """Fixture to mock auth settings globally"""
+    mock_settings = Settings(
+        auth0_domain="mock-domain.com",
+        auth0_audience="mock-audience",
+        auth0_management_id="mock-id",
+        auth0_management_secret="mock-secret",
+        auth0_algorithms=["RS256"],
+    )
+    mocker.patch("auth.config.get_settings", return_value=mock_settings)
+    mocker.patch("auth.management.get_settings", return_value=mock_settings)
+    return mock_settings
+
+@pytest.fixture
+def mock_auth_token(mocker):
+    """Fixture to mock authentication token"""
+    token = AccessTokenPayloadFactory.build(
+        sub="auth0|123456789",
+        biocommons_roles=["acdc/indexd_admin"],
+    )
+    mocker.patch("auth.validator.verify_jwt", return_value=token)
+    mocker.patch("auth.management.get_management_token", return_value="mock_token")
+    return token
+
+
+@pytest.fixture
+def auth_headers():
+    """Fixture to provide auth headers"""
+    return {"Authorization": "Bearer valid_token"}
+
+
+@pytest.fixture
+def mock_user_data():
+    """Fixture to provide mock user data"""
+    return {
+        "app_metadata": {
+            "services": [
+                {
+                    "id": "service1",
+                    "name": "Service 1",
+                    "status": "approved",
+                    "resources": [
+                        {"id": "resource1", "name": "Resource 1", "status": "approved"},
+                        {"id": "resource2", "name": "Resource 2", "status": "pending"},
+                    ],
+                },
+                {
+                    "id": "service2",
+                    "name": "Service 2",
+                    "status": "pending",
+                    "resources": [
+                        {"id": "resource3", "name": "Resource 3", "status": "pending"},
+                    ],
+                },
+            ]
+        }
+    }
+
+
+@pytest.mark.parametrize(
+    "endpoint",
+    [
+        "/me/services",
+        "/me/services/approved",
+        "/me/services/pending",
+        "/me/resources",
+        "/me/resources/approved",
+        "/me/resources/pending",
+        "/me/all/pending",
+    ],
+)
+
+def test_endpoints_require_auth(endpoint):
+    """Test that all endpoints require authentication"""
+    response = client.get(endpoint)
+    assert response.status_code == 401
+    assert response.json() == {"detail": "Not authenticated"}
+
+
+def test_get_all_services(
+    mock_auth_settings, mock_auth_token, auth_headers, mock_user_data, mocker
+):
+    """Test getting all services"""
+    mocker.patch(
```

Comment:
Can't we patch fetch_user_data to return the user data? I think that makes it a bit easier to follow what's happening in the test